helix-term/src/ui/popup.rs

```diff
@@ -95,14 +92,39 @@ impl<T: Component> Popup<T> {
         self
     }
 
-    /// Calculate the position where the popup should be rendered and return the coordinates of the
-    /// top left corner.
-    pub fn get_rel_position(&mut self, viewport: Rect, editor: &Editor) -> (u16, u16) {
+    pub fn scroll_half_page_down(&mut self) {
+        self.scroll_half_pages += 1;
+    }
+
+    pub fn scroll_half_page_up(&mut self) {
+        self.scroll_half_pages = self.scroll_half_pages.saturating_sub(1);
+    }
+
+    /// Toggles the Popup's scrollbar.
+    /// Consider disabling the scrollbar in case the child
+    /// already has its own.
+    pub fn with_scrollbar(mut self, enable_scrollbar: bool) -> Self {
+        self.has_scrollbar = enable_scrollbar;
+        self
+    }
+
+    pub fn contents(&self) -> &T {
+        &self.contents
+    }
+
+    pub fn contents_mut(&mut self) -> &mut T {
+        &mut self.contents
+    }
+
+    pub fn area(&mut self, viewport: Rect, editor: &Editor) -> Rect {
+        // trigger required_size so we recalculate if the child changed
+        let (width, height) = self
+            .required_size((viewport.width, viewport.height))
+            .expect("Component needs required_size implemented in order to be embedded in a popup");
+
         let position = self
             .position
-            .get_or_insert_with(|| editor.cursor().0.unwrap_or_default());
-
-        let (width, height) = self.size;
+            .unwrap_or_else(|| editor.cursor().0.unwrap_or_default());
```

Comment:
I think this should stay as `get_ir_insert` otherwise the popus move with the cursor. I also have some larger changes for this in #6417 IIRC since cursor calulation changes/is made more efficient there. 

I also wouldn't want to recompile teh cursor position all the time (which is slow)

Reply:
I tried to remove the mutation of `area` function. but missed that. already reverted